plugwise/smile.py

```diff
@@ -298,23 +298,23 @@ async def connect(self):
 
         return True
 
-    async def _smile_detect_legacy(self, result, dsmrmain):
+    async def _smile_detect_legacy(self, result, dsmrmain) -> tuple[str, str]:
         """Helper-function for _smile_detect()."""
-        network = result.find(".//module/protocols/master_controller")
+        network: etree | None = result.find(".//module/protocols/master_controller")
 
         # Assume legacy
         self._smile_legacy = True
         # Try if it is an Anna, assuming appliance thermostat
-        anna = result.find('.//appliance[type="thermostat"]')
+        anna: etree | None = result.find('.//appliance[type="thermostat"]')
         # Fake insert version assuming Anna
         # couldn't find another way to identify as legacy Anna
-        version = "1.8.0"
-        model = "smile_thermo"
+        version: str = "1.8.0"
+        model: str = "smile_thermo"
         if anna is None:
             # P1 legacy:
             if dsmrmain is not None:
                 try:
-                    status = await self._request(STATUS)
+                    status: etree = await self._request(STATUS)
```

Comment:
Typing should not be needed here, the problem is that the `_request` method isn't typed.